lib/Connection.js

```diff
@@ -108,6 +113,7 @@ class Connection extends EventEmitter {
   }
 
   _onClose() {
+    this._closeCallback();
```

Comment:
1. this should be the last line in the method
2. for the sake of cleanliness, let's check that delegate is set:
```js
if (this._closeCallback)
  this._closeCallback();
```